### PR TITLE
CE-2410 Refactor insights, unconverted infoboxes type and templatedra…

### PR DESCRIPTION
**Pull request by kamilkoterba**
…ft to remove dependency if any is switched off

https://wikia-inc.atlassian.net/browse/CE-2410

@Wikia/community-engineering

### CODE REVIEW

File: extensions/wikia/TemplateDraft/ApprovedraftAction.php [hunk 1]
```diff
@@ -96,10 +96,13 @@ private function approveDraft( Title $draftTitle ) {
 		$draftPage = WikiPage::newFromID( $draftTitle->getArticleID() );
 		$draftPage->doDeleteArticle( wfMessage( 'templatedraft-draft-removal-summary' )->inContentLanguage()->plain() );
 
-		// Update Insights list
-		$model = InsightsHelper::getInsightModel( InsightsUnconvertedInfoboxesModel::INSIGHT_TYPE );
-		if ( $model instanceof InsightsQuerypageModel ) {
-			$model->updateInsightsCache( $parentTitle->getArticleID() );
+		// Update Insights list if enabled
+		global $wgEnableInsightsInfoboxes;
```
Comment (lukaszkonieczny) [inline extensions/wikia/TemplateDraft/ApprovedraftAction.php]: Please move global var to the top of this method

**Comment by lukaszkonieczny** [comment]
lgtm, but please check unit tests

File: extensions/wikia/TemplateDraft/ApprovedraftAction.php [hunk 2]
```diff
@@ -96,10 +98,12 @@ private function approveDraft( Title $draftTitle ) {
...
 		$draftPage->doDeleteArticle( wfMessage( 'templatedraft-draft-removal-summary' )->inContentLanguage()->plain() );
 
-		// Update Insights list
-		$model = InsightsHelper::getInsightModel( InsightsUnconvertedInfoboxesModel::INSIGHT_TYPE );
-		if ( $model instanceof InsightsQuerypageModel ) {
-			$model->updateInsightsCache( $parentTitle->getArticleID() );
+		// Update Insights list if enabled
+		if ( $wgEnableInsightsInfoboxes ) {
+			$model = InsightsHelper::getInsightModel( InsightsUnconvertedInfoboxesModel::INSIGHT_TYPE );
+			if ( $model instanceof InsightsQuerypageModel ) {
```
Comment (drsnyder) [inline extensions/wikia/TemplateDraft/ApprovedraftAction.php]: Can we have `getInsightModel` always return an object that satisfies an interface that includes `updateInsightsCache`? 

Also it looks like the introduction of `$wgEnableInsightsInfoboxes` is leaking implementation details from `InsightsHelper` into this class. Could that behavior be encapsulated in `updateInsightsCache`?

**Comment by kamilkoterba** [comment]
Not possible for now, but I'll consider refactoring it to use dependency injection that we discussed once we move to insights refactoring.

**Pull request merged**